rust/src/haci/issuance_service_client.rs

```diff
@@ -34,13 +41,15 @@ struct NewIssuanceResponse {
 #[derive(Debug, Serialize, Deserialize, uniffi::Record)]
 pub struct CheckStatusResponse {
     state: String,
-    openid_credential_offer: String,
+    openid_credential_offer: Option<String>,
+    proofing_url: Option<String>,
 }
 
 #[derive(uniffi::Object)]
 pub struct IssuanceServiceClient {
     client: HaciHttpClient,
     base_url: String,
+    endpoints: RwLock<Option<HashMap<String, String>>>,
```

Comment:
Locks are not ideal, particularly across FFI in the case that they are poisoned. There's a few places in the code that can panic due to the use of RwLocks here.

I think a `tokio::sync::OnceCell` would be a better option here:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=39e09f0dd0f2e28157b2cc77593ad38b